I found that a 4pm or other non midnight required buffering. (#16182)
home-assistant/home-assistant.github.io: source/_integrations/history_stats.markdown

{% endraw %}

> **Next 4pm **: from 4pm today until 4pm tomorrow.

Reviewer: ```suggestion
**Next 4 pm**: 24 hours, from the last 4 pm till the next 4 pm. If it hasn't been 4 pm today, that would be 4 pm yesterday until 4 pm today. If it is already past 4 pm today, it will be 4 pm today until 4 pm tomorrow.
```

Author: Just wondering if it's worth mentioning the 8 hour buffer that would need changed if the start time is changed?

Perhaps ... 

**Next 4 pm**: 24 hours, from the last 4 pm till the next 4 pm. If it hasn't been 4 pm today, that would be 4 pm yesterday until 4 pm today. If it is already past 4 pm today, it will be 4 pm today until 4 pm tomorrow. If changing the start time, then add or subtract to the 8 hour buffer to match the next midnight.

Author: Not sure if I add this suggestion via a 'suggested change' I get "Authors can't do this" ...

{% raw %}

```yaml

>     end: '{{ (now().replace(minute=0,second=0) + timedelta(hours=8)).replace(hour=16) }}'

Reviewer: So why the juggling with all the hours? And not get today's 4pm + 1 day?

`{{ (now().replace(hour=16, minute=0, second=0) + timedelta(days=1)) }}`

Is there a specific reason for your approach?

>     end: '{{ (now().replace(minute=0,second=0) + timedelta(hours=8)).replace(hour=16) }}'

Reviewer: ```suggestion
    end: "{{ (now().replace(minute=0,second=0) + timedelta(hours=8)).replace(hour=16) }}"
```

> **Next 4 pm**: 24 hours, from the last 4 pm till the next 4 pm. If it hasn't been 4 pm today, that would be 4 pm yesterday until 4 pm today. If it is already past 4 pm today, it will be 4 pm today until 4 pm tomorrow.

Reviewer: ```suggestion
**Next 4 pm**: 24 hours, from the last 4 pm till the next 4 pm. If it hasn't been 4 pm today, that would be 4 pm yesterday until 4 pm today. If it is already past 4 pm today, it will be 4 pm today until 4 pm tomorrow. When changing the start time, then add or subtract to the 8-hour buffer to match the next midnight.
```

{% endraw %}

> **Next 4pm **: ends today at 00:00, lasts 30 days. Easy one.

Reviewer: - 4pm does not appear to be part of the example below.
- Please remove "Easy one.", it doesn't add anything and what may be easy for you, might not be easy for someone else.
